front/src/common/BootstrapSNCF/ModalSNCF/ModalProvider.tsx

```diff
@@ -58,7 +58,8 @@ export const ModalSNCF: FC = () => {
         isOpen &&
         modalRef.current &&
         !(event.target as HTMLElement).classList.contains('no-close-modal') &&
-        !modalRef.current.contains(event.target as HTMLElement)
+        !modalRef.current.contains(event.target as HTMLElement) &&
+        !modalRef.current?.classList.contains('no-close-modal')
       ) {
         closeModal();
       }
```

Comment:
Could we also use your new hook to remove the logic here ? Maybe we will have to keep the logic for the close-modal though

Reply:
fix